### DISCUSSION TITLE
I need val losses in the output file log.json but i can't get them

### DISCUSSION THREAD

**Discussion by AntonioVispi** [open-mmlab/mmclassification#1120] [General]
Greetings,
in order to get the loss graphs during training, I would need both the train losses, which appear regular, and the val losses. I have trouble getting the latter to appear. I have seen in past issues, and I understood that to do this I have to change the setting from 
`workflow = [('train', 1)]` to `workflow = [('train', 1), ('val', 1)]`, but for some reason, this gives me an Error.
Have you got any idea to fix it?

This is the part where I changed the workflow cfg.
![issue_1](https://user-images.githubusercontent.com/102518682/197226460-954117a0-abe4-4088-aafa-ed58c68c6f6b.png)
This is the error that arises
![issue_2](https://user-images.githubusercontent.com/102518682/197226877-90cbddfb-2db8-41e8-8d0f-5e494eb0a556.png)

Thanks in advance.

**Comment by AntonioVispi** [comment]
I solved it! 
I added these lines to the train command.
`import copy
val_dataset = copy.deepcopy(cfg.data.val)`
`val_dataset.pipeline = cfg.data.train.pipeline`
`datasets.append(build_dataset(val_dataset))`

Thank you anyway.
